plugins/guests/windows/cap/public_key.rb

```diff
@@ -6,81 +6,57 @@ module Cap
       class PublicKey
         def self.insert_public_key(machine, contents)
           if machine.communicate.is_a?(CommunicatorWinSSH::Communicator)
-            winssh_insert_public_key(machine, contents)
+            contents = contents.strip
+            modify_authorized_keys machine do |keys|
+              if !keys.include?(contents)
+                keys << contents
+              end
+            end
           else
             raise Vagrant::Errors::SSHInsertKeyUnsupported
           end
         end
 
         def self.remove_public_key(machine, contents)
           if machine.communicate.is_a?(CommunicatorWinSSH::Communicator)
-            winssh_remove_public_key(machine, contents)
+            modify_authorized_keys machine do |keys|
+              keys.delete(contents)
+            end
           else
             raise Vagrant::Errors::SSHInsertKeyUnsupported
           end
         end
 
-        def self.winssh_insert_public_key(machine, contents)
+        def self.modify_authorized_keys(machine)
```

Comment:
rename it to `winssh_modify_authorized_keys(machine)`?

the parameters cannot stay the same because that method is not really the same as `winssh_insert_public_key`. its used to implement the old `winssh_insert_public_key` and `winssh_remove_public_key`.